Library/ViewModels/CommentsViewModel.swift

```diff
@@ -197,10 +197,20 @@ public final class CommentsViewModel: CommentsViewModelType,
       .ksr_debounce(.nanoseconds(0), on: AppEnvironment.current.scheduler)
 
     self.loadCommentsAndProjectIntoDataSource = Signal.merge(
-      // If we start off with no comments, show an empty state.
-      commentsAndProject.take(first: 1),
-      // Subsequent empty emissions (pull to refresh) should be ignored until replaced.
-      commentsAndProject.skip(first: 1).filter { comments, _ in comments.isEmpty == false }
+      // Allow empty arrays from the first emission.
+      Signal.zip(comments, initialProject)
+        .filter { $0.0.isEmpty }
+        .map { ($0.0, $0.1) }
+        .map { comments, project in (comments, project, false) },
+      // Continue to paginate normally.
+      commentsAndProject
+        .filter { comments, _ in comments.isEmpty == false }
+        .map { comments, project in (comments, project, false) },
+      // If there are errors emit empty comments array, project and error boolean.
+      Signal.combineLatest(errors, hasRequestedNextPage)
+        .filter(second >>> isFalse)
+        .map(first)
+        .withLatestFrom(initialProject).map { ([], $1, true) }
```

Comment:
This is quite confusing for me, not sure if I follow it, but wouldn't be easier to just replace changes in this VM with:
```
    let showErrorLoadingForInitialComments = Signal.zip(initialLoadOrReload.signal,
                                                        errors.signal).mapConst(true)
```
and then create an output that just reloads the table view with an error state?

I'm not 100% of the side effects for these changes, that's all, because it touches to empty state, as well as the error states on pagination. Best to keep all of that separate imo.

Reply:
The error state would show if there is an error when you pull to refresh.

Reply:
I think that's ok because pull to refresh is like loading the first page. cc @singhhari